### DISCUSSION TITLE
Navigator cache for dotnet.wasm

### DISCUSSION THREAD

**Discussion by giraudremy** [unoplatform/uno#14310] [General]
Hi,

I have an uno Wasm application, and I compile it in AOT. The build generate a "dotnet.wasm" file which is over 100MB. To improve the application loading time, I added http headers to my Web server (expires max;add_header Cache-Control "public").
It works well on .clr files, but I always have the dotnet.wasm file which is systematically downloaded from the web server.

Is it possible to force caching this file?

Thank for reply

**Comment by jeromelaban** [comment]
Thanks for the question. Browsers are not configurable for this, though there are few options for this:
- Make sure that you have compression enabled for your server. The uno.wasm.bootstrap generates in-place compressed files (`.br`) to help for the size that you can use.
- Ensure that [the linker is running properly](https://platform.uno/docs/articles/features/using-il-linker-webassembly.html), and removing unused code wherever possible
- [Use Profiled AOT](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/runtime-execution-modes.html#profile-guided-aot) to only AOT compile the most used parts of your app.

**Comment by giraudremy** [comment]
Thank for reply.

To enable compression, it's web server configuration or it is option to set in csproj ? I'am using dotnet serve as web server, I have to launch it with -z|--gzip or -b|--brotli option ?

I'm going to try to activate the AOT profile again but I wasn't able to complete it last time.

Thanks for your help

**Reply by jeromelaban** [reply]
Compressing using on-the-fly methods is not particularly good for large files, as it will take a long time to generate. We do not explicitly support dotnet serve as file server either, as it does not provide support for pre-compressed files.

You'll need to use another server, like ones [we document here](https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html).

**Reply by giraudremy** [reply]
Thank for the link, I will try to migrate to nginx with brotli. Do you have docker image to host wasm application with nginx ? I found only build Dockerfile on https://github.com/unoplatform/docker/blob/master/
If not I will create my own.

Thanks

**Reply by jeromelaban** [reply]
We do not have images for serving apps at this time, but if you have steps to create one, we'll be glad to take a contribution to https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html!

**Reply by giraudremy** [reply]
Hi,

I'm creating a Docker image based on nginx with Brotli enabled and the configuration provided in the link. On my Wasm application that I generated over a year ago, it works very well, the compressed files are properly returned and cached by the browser because the `.br` files are in the same location as the uncompressed version.

Now on my updated application, I see that the compressed files are in `_compressed_br` and `_compressed_gz` folders and they don't have a `.br` extension for example. Is there a specific configuration to be done at the nginx level to serve the compressed files?

**Reply by giraudremy** [reply]
Ok I see the documentation, this because I have a web.config in my project with _compressed_br string.
I will use WasmShellCompressionLayoutMode to force inplace.

**Comment by giraudremy** [comment]
Hi,

I am happy to share with you my Docker image. This image is built upon NGINX and incorporates the Brotli module, enabling it to host a Wasm application. The NGINX configuration used in this setup is inspired by the recommendations provided by Uno.
[DockerNginxBrotli.zip](https://github.com/unoplatform/uno/files/13419870/DockerNginxBrotli.zip)
